tests/core/test_seeder.py

```diff
@@ -33,6 +33,17 @@ def generate_test_combs() -> List[Tuple[bool, str, dns.rdatatype.RdataType]]:
     return output
 
 
+def get_dns_port(dns_server: DNSServer, use_tcp: bool, target_addr: str) -> int:
+    if use_tcp:
+        assert dns_server.tcp_server is not None
+        tcp_sockets = dns_server.tcp_server.sockets
+        wanted_addr = "::" if target_addr == "::1" else "0.0.0.0"
+        if tcp_sockets[0].getsockname()[0] == wanted_addr:
+            return int(tcp_sockets[0].getsockname()[1])
+        return int(tcp_sockets[1].getsockname()[1])
+    return dns_server.udp_dns_port
```

Comment:
this function looks suspicious. why is the `use_tcp` path not looking at teh `tcp_dns_port` field?
it would also seem simpler to just have this function always return both ports instead of having the caller pick one

Reply:
and UDP always gets the same port for v4 and v6?

Reply:
Yep